### ISSUE TITLE
A suggestion for autocreation of modman file

### ISSUE THREAD

**Issue by waynetheisinger** [colinmollenhour/modman#68]
Hi Colin,

This is a suggestion, not an issue. How about an autocreate for the modman file? This would be very useful for when people retroactively create a modman file for their module rather than writing one when developing, maybe because it is a third-party module rather than their own.
### How it would work
1. `-bash$ modman automodman`
2. the script would traverse the directories below the one it was called from
3. at each directory level it would print the path and ask `include 'folder/path', [y] [n]`
4. if the answer was no but the folder contained files it would then loop through the files asking `include folder/path/filename [y] [n]`
5. If the answer was yes to either of the above it would create a simple modman line where both left and right  path matched (I think that if users are going have a different folder structure in their modman folder to that in Magento this would still save them half the work of creating a modman file.)
6. it would proceed to the next directory, either below its current level, or at a sibling level, or by returning to the parent and traversing down to the next in sequence. At each level the process 1 to 4 would be repeated
7. control C would break operation

What do you think?

**Comment by joh-klein** [comment]
nice

**Comment by bobbyburden** [comment]
I have another take on this as a ruby script [here](https://github.com/delegator/modman-utils/blob/master/modman-gen.rb).

**Comment by waynetheisinger** [comment]
@bobbyburden there are indeed many ways to remove the proverbial four legged feline mammal from its epidermis ;)

@joh-klein could be useful - I also considered the filename* glob pattern as way of achieving the adding of every file

ie

    include '/this/is/your/path/': [N][y][*]

which would create

    /this/is/your/path/*  /this/is/your/path/*

**Comment by colinmollenhour** [comment]
@waynetheisinger First of all, I am very impressed with your thoroughness. I think this is a good feature and it looks like you've implemented it really well and it doesn't break anything or fundamentally change anything. So, the only reason I haven't merged it is I haven't had time to test it out and look at it more closely, then I forgot about it.. So, sorry for that, I didn't mean to ignore your work.

I will add one thing in general is I like to stay away from building in too many features and options. For example, if a user wants to checkout/clone a repo that doesn't have a modman file, then they can already easily do this by:

```
$ cd .modman;
$ git clone ...
$ modman automodman
```

So I don't see the need in this case to complicate the checkout/clone command.

**Comment by waynetheisinger** [comment]
@colinmollenhour you're probably right about not needing to change the checkout/clone command

Also my fork has gone a little stale so if you are interested in doing a merge then I'll get mine up to date with your trunk and remove the checkout/clone code

There is also a small bug that I want to fix - it gives a false positive in the error checking for pre-existing paths in other modules, if comments in the other module, match the path that automodman is testing against - I think it should only throw the error for actual modman paths not those that are commented out....

I'll fix that too and then do another pull request

**Comment by colinmollenhour** [comment]
Yes, I am willing to merge this feature after said changes. Thanks!

**Comment by waynetheisinger** [comment]
@colinmollenhour I've made those changes and sent through another pull request... Thanks

**Comment by tmotyl** [comment]
just for the sake of completeness, here is a link to the tool already capable of generating modman files:
https://github.com/mhauri/generate-modman
